src/plugins/plugin.chapters.js

```diff
@@ -190,20 +190,16 @@ BookReader.prototype.getOpenLibraryRecord = function () {
     }
   };
 
-  $.ajax({ url: fetchUrlByBookId, dataType: 'jsonp' })
-    .then(data => {
-      if (data && data.length > 0) {
-        return data;
-      } else {
-      // try sourceid
-        return $.ajax({ url: `${baseURL}&source_records=ia:${this.bookId}`, dataType: 'jsonp' });
-      }
-    })
-    .then(data => {
-      if (data && data.length > 0) {
-        setUpChapterMarkers(data[0]);
-      }
-    });
+  let data = await $.ajax({ url: fetchUrlByBookId, dataType: 'jsonp' });
+
+  if (!data && data.length < 0) {
```

Comment:
Oh the logic for this isn't quite right; it should be this.

(If you're interested, negating an `&&` expression is called applying De Morgan's Law: https://en.wikipedia.org/wiki/De_Morgan%27s_laws#:~:text=not%20(A%20and%20B)%20%3D%20(not%20A)%20or%20(not%20B)%2C )

```suggestion
  if (!data || !data.length) {
```